api/app/zonal_stats.py

```diff
@@ -250,18 +253,33 @@ def calculate_stats(
     if group_by:
         zones_filepath = _group_zones(zones_filepath, group_by)
 
-    stats_input: FilePath | list = zones_filepath
+    zones_geojson: GeoJSON = get_json_file(zones_filepath)
+
+    if filter_by is not None:
+        key, value = filter_by
+        stats_input: list[Geometry] = [
+            f["geometry"]
+            for f in zones_geojson["features"]
+            if key in f["properties"].keys() and str(f["properties"][key]) == value
+        ]
+
+        if len(stats_input) == 0:
+            message = f"Property value '{value}' not found"
```

Comment:
Could we add the key in this message as well for debugging?

Reply:
included in [347b1a9](https://github.com/WFP-VAM/prism-app/pull/616/commits/347b1a9d47ff2e2c62a97386233710d5b9756371)